src/main/java/InvalidIndexException.java

```diff
@@ -0,0 +1,2 @@
+public class InvalidIndexException extends Exception {
+}
```

Comment:
Since you are identifying different kinds of exceptions, you should write javadoc comments to give the users a context of why this exception appears in your codebase.

Same for the above.